### ISSUE TITLE
[java]ThreadLocalFury has performance issues under virtual threads in jdk21

### ISSUE THREAD

**Issue by gongziyiyi** [apache/incubator-fury#1570]
## Is your feature request related to a problem? Please describe.
ThreadLocal has performance issues under virtual threads in jdk21

## Describe the solution you'd like
maybe use CarrierThreadLocal ??

**Comment by chaokunyang** [comment]
Will It create lots of Fury objects? For example, if we have million of virtual threads, will we have millions of Fury object?

**Comment by gongziyiyi** [comment]
@chaokunyang 
Yes. CarrierThreadLocal can fix it, but unfortunately, it has not been leaked for external use and can only be opened with -- add open at the moment

**Comment by chaokunyang** [comment]
We may need to add a new maven module to fury/java.

And I'm wondering whether can we use ThreadPoolFury, which use an object pool. This may be more friently to virtual thread
